### ISSUE TITLE
[Woo POS] Improvement to the "no internet connection" notification

### ISSUE THREAD

**Issue by iamgabrielma** [woocommerce/woocommerce-ios#13743]
The existing "no internet connection" notification doesn't work properly when detecting if connection has been recovered, which leads to the notification still showing up despite having internet connection.

![Simulator Screenshot - iPad Pro (12 9-inch) (6th generation) - 2024-08-27 at 10 07 18](https://github.com/user-attachments/assets/db977136-6525-4ca2-8291-fd39a1f00c2b)

A potential improvement could be to add tappable behaviour to the notification: If tapped, we can directly check if there's connection, rather than relying on observing the reachability state.

This fits Apple's recommendation of _"do not attempt to guess whether network service is available, and do not cache that determination [...] The SCNetworkReachability API is not intended for use as a preflight mechanism for determining network connectivity. You determine network connectivity by attempting to connect."_

To replicate:
- In POS mode, turn off the internet
- Turn it back on. 
- Observe that the notification state is not updated immediately.

**Comment by dangermattic** [comment]
Thanks for reporting! 👍

**Comment by malinajirka** [comment]
@iamgabrielma Is this still valid? If it is and you believe it's wroth spending time on, assign a priority please. Thanks!

**Comment by iamgabrielma** [comment]
Personally I don't think it's worth to invest additional time on it, and if we do, it should be done both for app and POS as we reuse the connectivity components. But yet again, nobody has complained, and is easily resolved just by tapping around the device once network connection has been restored.

cc: Just looping @staskus as per initial implementation. Thoughts?

**Comment by staskus** [comment]
@iamgabrielma I agree it's not perfect, it doesn't react immediately to reachability changes. I also agree it's not worth spending time improving it.
